src/iterable-functions.php

```diff
@@ -136,6 +136,21 @@ function iterable_values(iterable $iterable): iterable
     return iterable($iterable)->values();
 }
 
+/**
+ * Split an iterable into chunks
+ *
+ * @param iterable<TKey, TValue> $iterable
+ *
+ * @return iterable<array<TKey, TValue>>
```

Comment:
I'd prefer 

```suggestion
 * @return iterable<iterable<TKey, TValue>>
```

Since original input has no constraint on the key type, the `array` might not support it.

Reply:
Good point, thx!